jdisc_http_service/src/main/java/com/yahoo/container/logging/LogFileHandler.java

```diff
@@ -110,320 +50,426 @@ private void flushIfOld(long age, TimeUnit unit) {
+                i++;
+                continue;
+            } // skip spaces
+            int j = i; // start of string
+            i = times.indexOf(' ', i);
+            if (i == -1) i = times.length();
+            if (times.charAt(j) == '.' && times.substring(j, i).equals("...")) { // ...
+                etc = true;
                 break;
             }
-        }
-        if (next == 0) { // didn't find one -- use 1st time 'tomorrow'
-            next = rotationTimes[0]+lengthOfDayMillis-nowTod + now;
+            list.add(Long.valueOf(times.substring(j, i)));
         }
 
-        return next;
-    }
+        int size = list.size();
+        long[] longtimes = new long[size];
+        for (i = 0; i < size; i++) {
+            longtimes[i] = list.get(i)   // pick up value in minutes past midnight
+                           * 60000;                          // and multiply to get millis
+        }
 
-    void waitDrained() {
-        while(! logQueue.isEmpty()) {
-            try {
-                Thread.sleep(1);
-            } catch (InterruptedException e) {
+        if (etc) { // fill out rest of day, same as final interval
+            long endOfDay = 24 * 60 * 60 * 1000;
+            long lasttime = longtimes[size - 1];
+            long interval = lasttime - longtimes[size - 2];
+            long moreneeded = (endOfDay - lasttime) / interval;
+            if (moreneeded > 0) {
+                int newsize = size + (int) moreneeded;
+                long[] temp = new long[newsize];
+                for (i = 0; i < size; i++) {
+                    temp[i] = longtimes[i];
+                }
+                while (size < newsize) {
+                    lasttime += interval;
+                    temp[size++] = lasttime;
+                }
+                longtimes = temp;
             }
         }
-        flush();
-    }
 
-    private void checkAndCreateDir(String pathname) {
-      int lastSlash = pathname.lastIndexOf("/");
-      if (lastSlash > -1) {
-          String pathExcludingFilename = pathname.substring(0, lastSlash);
-          File filepath = new File(pathExcludingFilename);
-          if (!filepath.exists()) {
-            filepath.mkdirs();
-          }
-      }
+        return longtimes;
     }
 
     /**
-     * Force file rotation now, independent of schedule.
+     * Only for unit testing. Do not use.
      */
-    void rotateNow () {
-        rotate.set(true);
+    String getFileName() {
+        return logThread.fileName;
     }
 
-    // Throw InterruptedException upwards rather than relying on isInterrupted to stop the thread as
-    // isInterrupted() returns false after interruption in p.waitFor
-    private void internalRotateNow() {
-        // figure out new file name, then
-        // use super.setOutputStream to switch to a new file
+    /**
+     * Handle logging and file operations
+     */
+    static class LogThread<LOGTYPE> extends Thread {
+        long lastFlush = 0;
```

Comment:
Make `private`

Reply:
Currently there's a non-static method in this class. Will see if we can resolve this.

Reply:
Currently there's a test referencing a method in this class. Let's see if we can untangle after this refactoring (as part of this pr)